spanner/cloud-client/src/main/java/com/example/spanner/SpannerSample.java

```diff
@@ -1473,9 +1503,371 @@ static void queryWithQueryOptions(DatabaseClient dbClient) {
+      }
+    } catch (ExecutionException e) {
+      throw (SpannerException) e.getCause();
+    } catch (InterruptedException e) {
+      throw SpannerExceptionFactory.propagateInterrupt(e);
+    }
+  }
+  // [END spanner_cancel_create_backup]
+
+  // [START spanner_list_backup_operations]
+  static void listBackupOperations(InstanceAdminClient instanceAdminClient, DatabaseId databaseId) {
+    Instance instance = instanceAdminClient.getInstance(databaseId.getInstanceId().getInstance());
+    // Get create backup operations for the sample database.
+    String filter =
+        String.format(
+            "(metadata.database:%s) AND "
+                + "(metadata.@type:type.googleapis.com/"
+                + "google.spanner.admin.database.v1.CreateBackupMetadata)",
+            databaseId.getName());
+    Page<Operation> operations = instance.listBackupOperations(Options.filter(filter));
+    for (Operation op : operations.iterateAll()) {
+      try {
+        CreateBackupMetadata metadata = op.getMetadata().unpack(CreateBackupMetadata.class);
+        System.out.println(
+            String.format(
+                "Backup %s on database %s pending: %d%% complete",
+                metadata.getName(),
+                metadata.getDatabase(),
+                metadata.getProgress().getProgressPercent()));
+      } catch (InvalidProtocolBufferException e) {
+        // The returned operation does not contain CreateBackupMetadata.
+        System.err.println(e.getMessage());
+      }
+    }
+  }
+  // [END spanner_list_backup_operations]
+
+  // [START spanner_list_database_operations]
+  static void listDatabaseOperations(
+      InstanceAdminClient instanceAdminClient,
+      DatabaseAdminClient dbAdminClient,
+      DatabaseId databaseId) {
+    Instance instance = instanceAdminClient.getInstance(databaseId.getInstanceId().getInstance());
+    // Get optimize restored database operations for the sample database.
+    String filter = String.format("(name:%s) AND (metadata.@type:type.googleapis.com/"
+                    + "google.spanner.admin.database.v1.OptimizeRestoredDatabaseMetadata)",
+                    databaseId.getDatabase());
+    for (Operation op : instance.listDatabaseOperations(Options.filter(filter)).iterateAll()) {
+      try {
+        OptimizeRestoredDatabaseMetadata metadata =
+            op.getMetadata().unpack(OptimizeRestoredDatabaseMetadata.class);
+        DatabaseId restoredDbId = DatabaseId.of(
+            databaseId.getInstanceId(),
+            metadata.getName().substring(metadata.getName().lastIndexOf('/') + 1));
+        Database restoredDb = new Database(restoredDbId, State.UNSPECIFIED, dbAdminClient);
+        if (restoredDb.exists()) {
+          RestoreInfo restore = restoredDb.getRestoreInfo();
+          if (restore != null) {
+            System.out.println(
+                String.format(
+                    "Database %s restored from backup %s is optimized.",
```

Comment:
I think to keep things simpler we can just do:

`"Database %s restored from backup is %s%% optimized"`

This is what we ended up doing in other languages. This way, you don't need to get the `restoreDb` or the `restore` info, etc.

Reply:
done